lib/rules/indent.js

```diff
@@ -970,33 +970,33 @@ module.exports = {
             const parenStack = [];
             const parenPairs = [];
 
-            tokens.forEach(nextToken => {
+            for (const nextToken of tokens) {
```

Comment:
this category of change is not necessarily an improvement - the iterator protocol can be much slower than array iteration.

Reply:
Is it actually slower than calling a function in every loop? Both of them are slower than a regular `for` loop, but `for-of`’s performance has been improved since its introduction 6 years ago.

Reply:
I don’t know; I haven’t ran benchmarks - but it has to invoke a Symbol.iterator method and deal with iteration objects, or else it has to optimize those away - array iteration is always faster, or as fast, depending on which optimizations are present.

My personal opinion, and that of the Airbnb style guide, is that loops are always to be avoided and that array iteration methods are always preferred, even when using reduce.

Reply:
`for-of` slower then `for` a little bit, but twice faster then `forEach`
https://jsbench.github.io/#cec5fd6009b622c72311e8f447e101be

Reply:
That doesn't seem right at all. In v8 (ie, node), since 2018, array iteration methods are 100% identically as fast as a for loop, since they JIT to the same instructions.

Reply:
Do you have any links :)? Maybe `v8` team wrote something? Or other benchmarks? For me and benchmark it is seems right. You call a function in one case, and iterate array in second without calling a function, so yes it is faster.